docker_entrypoint.sh

```diff
@@ -40,189 +67,304 @@ ADV_TOR_MAXSOCKSCONNS=$(yq e '.advanced.tor.maxsocksconns' ${BITMONERO_DIR}/star
  #fi
 fi
  
 #CUSTOM NODES config:
-echo -e "\n# CUSTOM NODES" >> $new_conf_template
+echo -e "\n# CUSTOM NODES" >> $conf_template_new
 i=1
 num_custom_peers=$(yq e '.advanced.p2p.peer[]|length' ${BITMONERO_DIR}/start9/config.yaml | wc -l)
 while [[ $i -le $num_custom_peers ]] ; do
  peer_hostname=$(yq e '.advanced.p2p.peer[$i].hostname' ${BITMONERO_DIR}/start9/config.yaml)
  peer_port=$(yq e '.advanced.p2p.peer[$i].port' ${BITMONERO_DIR}/start9/config.yaml)
  peer_priority=$(yq e '.advanced.p2p.peer[$i].prioritynode' ${BITMONERO_DIR}/start9/config.yaml)
  if [ "$ADV_P2P_STRICTNODES" = "true" ] ; then
-  echo "add-exclusive-node=$peer_hostname:$peer_port" >> $new_conf_template
+  echo "add-exclusive-node=$peer_hostname:$peer_port" >> $conf_template_new
  else
   if [ "$peer_priority" = "true" ] ; then
-   echo "add-priority-node=$peer_hostname:$peer_port" >> $new_conf_template
+   echo "add-priority-node=$peer_hostname:$peer_port" >> $conf_template_new
   else
-   echo "add-peer=$peer_hostname:$peer_port" >> $new_conf_template
+   echo "add-peer=$peer_hostname:$peer_port" >> $conf_template_new
   fi
-  echo "" >> $new_conf_template
+  echo "" >> $conf_template_new
  fi
  i=$(expr $i + 1)
 done
 
 #If the user has enabled BTCPayServer integration, send block notifications there
 if [ "$INT_ANN_BLOCKS_TO_BTCPAY" == "true" ] ; then
- btcpay_integration="block-notify='/usr/bin/curl -X GET \"http://btcpayserver.embassy:23001/monerolikedaemoncallback/block?cryptoCode=xmr&hash=%s\"'"
- echo -e "\n# BLOCK NOTIFICATIONS\n${btcpay_integration}" >> $new_conf_template
+ btcpay_integration='block-notify=/usr/bin/curl -so /dev/null -X GET http://btcpayserver.embassy:23001/monerolikedaemoncallback/block?cryptoCode=xmr&hash=%s'
+ echo -e "\n# BLOCK NOTIFICATIONS\n${btcpay_integration}" >> $conf_template_new
 fi
 
-mv $new_conf_template $new_conf
 
+# Done setting monero.conf options.  Now process monero-wallet-rpc.conf:
+
+#Replace placeholders in monero-wallet-rpc.conf template with their corresponding values
+sed -i "s/MONERO_RPC_PORT_WALLET_RPC/$MONERO_RPC_PORT_WALLET_RPC/" $wallet_rpc_conf_new
+sed -i "s/MONERO_RPC_PORT/$MONERO_RPC_PORT/" $wallet_rpc_conf_new
+sed -i "s/MONEROD_BIND_IP/$MONEROD_BIND_IP/" $wallet_rpc_conf_new
+sed -i "s/MONERO_WALLET_DIR/$MONERO_WALLET_DIR_ESC/" $wallet_rpc_conf_new
+sed -i "s/MONERO_WALLET_RPC_LOG/$MONERO_WALLET_RPC_LOG_ESC/" $wallet_rpc_conf_new
+sed -i "s/RPC_WALLET_USERNAME/$RPC_WALLET_USERNAME/" $wallet_rpc_conf_new
+sed -i "s/RPC_WALLET_PASSWORD/$RPC_WALLET_PASSWORD/" $wallet_rpc_conf_new
+sed -i "s/RPC_USERNAME/$RPC_USERNAME/" $wallet_rpc_conf_new
+sed -i "s/RPC_PASSWORD/$RPC_PASSWORD/" $wallet_rpc_conf_new
+#Comment RPC logins options if credentials have been disabled
+if [ "$RPC_WALLET_CREDENTIALS" == "disabled" ] ; then
+ sed -i "s/^\(daemon-login=.*\)$/#\1/" $wallet_rpc_conf_new
+fi
+if [ "$RPC_CREDENTIALS" == "disabled" ] ; then
+ sed -i "s/^\(rpc-login=.*\)$/#\1/" $wallet_rpc_conf_new
```

Comment:
maybe just delete this line instead of replacing it with nothing

Reply:
changed approach